aws-opentelemetry-distro/tests/amazon/opentelemetry/distro/test_instrumentation_patch.py

```diff
@@ -156,10 +251,76 @@ def _do_extract_sqs_attributes() -> Dict[str, str]:
     return _do_extract_attributes(service_name, params)
 
 
-def _do_extract_attributes(service_name: str, params: Dict[str, str]) -> Dict[str, str]:
+def _do_bedrock_on_success() -> Dict[str, str]:
+    service_name: str = "bedrock"
+    result: Dict[str, Any] = {"guardrailId": _BEDROCK_GUARDRAIL_ID}
+    return _do_on_success(service_name, result)
+
+
+def _do_extract_attributes_bedrock(service, operation=None) -> Dict[str, str]:
+    params: Dict[str, Any] = {
+        "agentId": _BEDROCK_AGENT_ID,
+        "dataSourceId": _BEDROCK_DATASOURCE_ID,
+        "knowledgeBaseId": _BEDROCK_KNOWLEDGEBASE_ID,
+        "guardrailId": _BEDROCK_GUARDRAIL_ID,
+        "modelId": _BEDROCK_MODEL_ID,
+    }
+    return _do_extract_attributes(service, params, operation)
+
+
+def _do_on_success_bedrock(service, operation=None) -> Dict[str, str]:
+    result: Dict[str, Any] = {
+        "agentId": _BEDROCK_AGENT_ID,
+        "dataSourceId": _BEDROCK_DATASOURCE_ID,
+        "knowledgeBaseId": _BEDROCK_KNOWLEDGEBASE_ID,
+        "guardrailId": _BEDROCK_GUARDRAIL_ID,
+        "modelId": _BEDROCK_MODEL_ID,
+    }
+    return _do_on_success(service, result, operation)
+
+
+def _do_extract_bedrock_agent_runtime_attributes() -> Dict[str, str]:
+    service_name: str = "bedrock-agent-runtime"
+    params: Dict[str, str] = {"agentId": _BEDROCK_AGENT_ID, "knowledgeBaseId": _BEDROCK_KNOWLEDGEBASE_ID}
+    return _do_extract_attributes(service_name, params)
+
+
+def _do_extract_bedrock_runtime_attributes() -> Dict[str, str]:
```

Comment:
Fine to have a separate _do_extract_bedrock_runtime_attributes as it's different from the others, but I would still add agentId, etc to the params to show they are not used.